my_Enumerable.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:double) unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:double) unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:double) unless block_given?
+
+    condition_met = []
+    my_each do |item|
+      condition_met << item if yield(item)
+    end
+    condition_met
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return false unless yield(item)
+      elsif pattern
+          return false unless pattern === item
+        else
+          return false if item.nil? || item == false
+        end
+      end
+    end
```

Comment:
Be careful with this, you have an extra `end` over here.

it should look like:
```

  def my_all?(pattern = nil)
    my_each do |item|
      if block_given?
        return false unless yield(item)
      elsif pattern
          return false unless pattern === item
      else
          return false if item.nil? || item == false
      end
    end
    true
  end
```